### DISCUSSION TITLE
Put parrot on player's shoulder

### DISCUSSION THREAD

**Discussion by ZakShearman** [Minestom/Minestom#1438] [Help] [answered]
How can I get the NBT of a parrot entity so I can put them onto a player's shoulder (`player.getEntityMeta().setLeftShoulderEntityData()`)?

**Comment by Omnyseus** [comment, marked as answer]
Well.. i've tried for a few hours now how to do it myself and had a dumb mistake from the beginning (used an additional "ShoulderEntityLeft" NBTCompound..). So what you need to do:

(in my case i spawn the parrot per command and pick it up when i interact with it)

``` kotlin 
  fun onInteract(event: PlayerEntityInteractEvent) {
        if (event.target.entityType != EntityType.PARROT) {
            return
        }

        val metadata = event.target.entityMeta as ParrotMeta
        val parrot = event.target

        event.player.entityMeta.leftShoulderEntityData = NBT.Compound(
            mapOf(
                //you need to have the id set to "minecraft:parrot", otherwise nothing will happen
                "id" to NBT.String("minecraft:parrot"),
                //optional Variant/Color for the Parrot, default is red i think
                "Variant" to NBT.Int(metadata.color.ordinal),
            )
        )

        //remove the original parrot if you want to (otherwise it will visually duplicate, one where it was before picking it up and one on your shoulder, which is not an entity anymore!)
        parrot.remove()
    }

**Reply by emortaldev** [reply]
Can you use any entity or does it just not show up

**Reply by Omnyseus** [reply]
It won't show up. Tried to test it with frogs and chickens cause they are also smol but nothing happens

**Reply by ZakShearman** [reply]
Do you know a way of making them not disappear when you fly?

**Reply by Omnyseus** [reply]
I think the client just predicts to drop it and clears its metadata. For other players it won't disappear.
I've tried to readd it every tick but that lead to flickering. I also tried to add additional metadata, like "PersitenceRequired" to true but that didn't have any impact either

**Reply by ZakShearman** [reply]
Yep for now I'm just removing it when they fly and adding it back after. Thx <3

**Reply by KorneIlles** [reply]
> It won't show up. Tried to test it with frogs and chickens cause they are also smol but nothing happens

but if you make a custome model entyti what is implement that Parrot feature maybe that's gonna work but this is just a theory :D
